includes/WikiPage.php

```diff
@@ -590,10 +590,12 @@ public function isCountable( $editInfo = false ) {
 	 *
 	 * The target will be fetched from the redirect table if possible.
 	 * If this page doesn't have an entry there, call insertRedirect()
+	 *
+	 * @param int $useMasterDb
 	 * @return Title|mixed object, or null if this page is not a redirect
 	 */
-	public function getRedirectTarget() {
-		if ( !$this->mTitle->isRedirect() ) {
+	public function getRedirectTarget( $useMasterDb = 0 ) {
```

Comment:
Just to be a stickler, you should probably use the same signature as other methods that accept this, e.g.:
```php
public function getRedirectTarget( $flags = 0 )
```
Especially since what you pass to ```isRedirect``` is not a boolean, but this bit field value.  Then on L607 you would check:
```php
$dbr = $flags & \Title::GAID_FOR_UPDATE ? ...
```
If you want to be clear about the only flag you care about you could do your bitwise just prior into a variable:
```php
$useMasterDb = $flags & \Title::GAID_FOR_UPDATE;
```